src/lib/core/CHIPTLV.h

```diff
@@ -443,5 +2195,119 @@ class DLL_EXPORT TLVUpdater
     const uint8_t * mElementStartAddr;
 };
 
+/**
+ * Provides an interface for TLVReader or TLVWriter to use memory other than a simple contiguous buffer.
```

Comment:
Are the read and write cases separable into two different interfaces here? Should they be?

Reply:
They should be, and in fact I started out with separate interfaces `TLVReader::BackingStore` and `TLVWriter::BackingStore`, but ran into trouble, if I recall correctly with `CircularTLVBuffer`.